src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const currentState = { ...state };
+
+  for (const action of actions) {
+    const { extraData, type, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        // eslint-disable-next-line no-const-assign
+        for (const key in currentState) {
+          delete currentState[key];
+        }
+        break;
+
+      case 'addProperties':
+        Object.assign(currentState, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+
+      default:
+        throw new Error('Error');
```

Comment:
The default case of the switch statement should not throw an error unless specified by the task requirements. It's better to handle unexpected action types gracefully, perhaps by continuing to the next iteration of the loop.